src/UI/Components/Diagram/Context/EventWrapper.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useContext, useEffect } from "react";
+import { updateInstancesToSend } from "../helpers";
+import { ActionEnum, EmbeddedEventEnum } from "../interfaces";
+import { ServiceEntityBlock } from "../shapes";
+import { CanvasContext } from "./Context";
+
+/**
+ * EventWrapper component
+ *
+ * This component is a higher-order component that wraps its children and provides event handling for all necessary communication from withing JointJS code to the Composer.
+ * It uses the CanvasContext to get and set various state variables.
+ *
+ * @props {React.PropsWithChildren} props - The properties that define the behavior and display of the component.
+ * @prop {React.ReactNode} children - The children components to be wrapped.
+ *
+ * @returns {React.FC<React.PropsWithChildren>} The EventWrapper component.
+ */
+export const EventWrapper: React.FC<React.PropsWithChildren> = ({
+  children,
+}) => {
+  const {
+    setStencilState,
+    setInstancesToSend,
+    setCellToEdit,
+    setDictToDisplay,
+    looseEmbedded,
+    setLooseEmbedded,
+  } = useContext(CanvasContext);
+
+  /**
+   * Handles the event triggered when there are loose embedded entities on the canvas.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleLooseEmbeddedEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+    const eventData: { kind: EmbeddedEventEnum; id: string } = JSON.parse(
+      customEvent.detail,
+    );
+    const newSet = new Set(looseEmbedded);
+
+    if (eventData.kind === "remove") {
+      newSet.delete(eventData.id);
+    } else {
+      newSet.add(eventData.id);
+    }
+    setLooseEmbedded(newSet);
+  };
+
+  /**
+   * Handles the event triggered when the user wants to see the dictionary properties of an entity.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleDictEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+
+    setDictToDisplay(JSON.parse(customEvent.detail));
+  };
+
+  /**
+   * Handles the event triggered when the user wants to edit an entity.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleEditEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+
+    setCellToEdit(customEvent.detail);
+  };
+
+  /**
+   * Handles the event triggered when the user made update to the instance cell
+   * With removed ability to edit the related instances, we need to assert first if the instance triggered the event is in the instancesToSend map(it could be removed from the canvas)
```

Comment:
becase there is no longer ability to do that in the composer

Reply:
Comments shouldn't be about an older state of the code, but about the current state. If someone new to the team reads this, they will not understand the comment because they don't have the history of the past year like we do. 

The better approach would be to say that editing related instances is not allowed, and add the reason why that is.

Reply:
modified